File: PortalOS/PortalOS.Web/src/types/cliente.ts
export const StatusCliente = {
  Ativo: 0,
  Inativo: 1
} as const;

export type StatusCliente = typeof StatusCliente[keyof typeof StatusCliente];

export interface Cliente {
  id: number;
  razaoSocial: string;
  cnpj: string;
  responsavel: string;
  telefone: string;
  emailResponsavel: string;
  emailFinanceiro: string;
  endereco: string;
  statusCliente: StatusCliente;
  clienteTotvs: boolean;
}

export interface CreateClienteRequest {
  razaoSocial: string;
  cnpj: string;
  responsavel: string;
  telefone: string;
  emailResponsavel: string;
  emailFinanceiro: string;
  endereco: string;
  statusCliente: StatusCliente;
  clienteTotvs: boolean;
}

export interface UpdateClienteRequest extends CreateClienteRequest {
  id: number;
}

File: PortalOS/PortalOS.Domain/Entities/Cliente.cs
using dNET.Domain.Entities;
using dNET.Domain.Utils.Attributes;
using PortalOS.Domain.ValueObjects;

namespace PortalOS.Domain.Entities
{
    public class Cliente : EntidadeBase
    {
        [NotEmpty]
        public string RazaoSocial { get; set; }

        [CpfCnpj]
        public string Cnpj { get; set; }

        public string Responsavel { get; set; }

        [NumberPhone]
        public string Telefone { get; set; }

        public string EmailResponsavel { get; set; }

        public string EmailFinanceiro { get; set; }

        public string Endereco { get; set; }

        public StatusCliente StatusCliente { get; set; }

        public bool ClienteTotvs { get; set; }
    }
}
